src/battle_script_commands.c

```diff
@@ -16383,3 +16317,78 @@ void BS_SetSnow(void)
     }
     gBattlescriptCurrInstr = cmd->nextInstr;
 }
+
+void BS_HandleMegaEvolution(void)
+{
+    NATIVE_ARGS(u8 battler, u8 caseId);
+
+    gActiveBattler = GetBattlerForBattleScript(cmd->battler);
```

Comment:
in preparation for #3262 let's use a local variable instead of gActiveBattler

Reply:
Weird thing is, I initially tried using a local variable but for somr reason it doubled the messages when when doing it for Ultra Burst, which is the first one I tried. I'll try to find the source of the issue once I get time.

Reply:
Maybe there was some downstream dependence on gActiveBattler. Let's hold off until that pr gets merged

Reply:
Sure, though the current test didn't pick up on the double message, so that's something to check with that PR as well.

Reply:
This was indeed the case, double messags are gone after the merge with upcoming.